### ISSUE TITLE
[BUG] - test_bitcoin should stop bitcoind on exit

### ISSUE THREAD

**Issue by bitcoin-tools** [bitcoin-tools/nodebuilder#725]
### Expected Behavior

don't leave the process running after finishing the test

### Actual Behavior

Core is left running after the test finishes/fails

### Steps to Reproduce

1. 
2. 
3. 

### Optional Logs

```
+ '[' true = true ']'
+ return 0
+ bitcoind --daemonwait
Error: Cannot obtain a lock on data directory /Users/runner/Library/Application Support/Bitcoin. Bitcoin Core is probably already running.
+ '[' 1 -eq 0 ']'
+ handle_error 1037
+ log_error 'Failed to run line 1037.'
++ get_cursor_column
+++ stty -g
stty: stdin isn't a terminal
++ stored_stty_settings=
+ '[' '' -gt 0 ']'
./nodebuilder: line 326: [: : integer expression expected
++ date +%Y-%m-%dT%H:%M:%S
+ printf '%s\n' '[2024-05-03T16:44:57] ERROR: Failed to run line 1037.'
[2024-05-03T16:44:57] ERROR: Failed to run line 1037.
+ handle_exit
+ '[' -n /var/folders/3m/p59k4qdj0f17st0gn2cmj3640000gn/T/tmp.uYY8N7D3R5 ']'
+ rm -rf -- /var/folders/3m/p59k4qdj0f17st0gn2cmj3640000gn/T/tmp.uYY8N7D3R5/
+ exit 1
```

**Comment by bitcoin-tools** [comment]
Resolved with #733.

**Comment by bitcoin-tools** [comment]
Duplicate of #732.
